### ISSUE TITLE
Duplicate text after submit

### ISSUE THREAD

**Issue by syntacticsugar** [TeamWertarbyte/material-ui-chip-input#13]
After typing text, example _"hello"_, and hitting enter, the chip properly displays, but the _'hello'_ text I typed still lingers immediately after the chip.  So it looks like this:

![screen shot 2016-09-28 at 10 23 19 pm](https://cloud.githubusercontent.com/assets/998068/18938913/37a3ad1c-85ca-11e6-88c8-9249adbe872f.png)

If this helps any, the 'ghost'/'phantom' text doesn't turn into a chip if I hit enter a second time.  If I add one character to the phantom text (and thereby changing it), I can hit enter a second time and turn the new string into a chip.  

I don't want this duplicate text at all.  Here's the code I had written:

```
<ChipInput
          //defaultValue={[]}
          fullWidth
          floatingLabelText="start typing"
          onChange={(chips) => handleChange(chips)}
 />
```

It's wrapped inside the `<MuiThemeProvider>`, by the way.  Apologies if I'm making a silly mistake.  Help is GREATLY appreciated!!!  Thank you!

**Comment by leMaik** [comment]
This is due to a race condition of some events that I try to work around in a pretty ugly way at the moment. I'll see if I can come up with a better solution.

Your usage of the compopnent is correct, it even _needs_ to be inside a `MuiThemeProvider`. :wink:

**Comment by leMaik** [comment]
@syntacticsugar (great username btw :laughing:) Does the commit above fix this issue for you? I'd publish a new version if so.

**Comment by syntacticsugar** [comment]
Wow, thank you for the fast response 😀 I'm about to get on the subway and
commute, but as soon as I get home I'll investigate! 😄

**Comment by syntacticsugar** [comment]
Hello!  I spent the past hour trying to get your latest commit to work, but I'm not sure I'm succeeding.  Here's what I did:

1.  `npm install --save https://github.com/TeamWertarbyte/material-ui-chip-input.git#bb730e97a37f7`
2.  got this error upon `npm start` : "Module not found: Error: Cannot resolve 'file' or 'directory' ../node_modules/material-ui-chip-input in  [path to my project/src]
3.  tried `npm install` again
4.  got same error #2 
5.  ran `ls node_modules | grep -ir "material*"` to ensure I **did** have the module
6.  copied via `cp node_modules/material-ui/chip-input/src/ChipInput.js myProject/src/ChipInput.js`
7.  in my JS file, I made sure to import correctly `import ChipInput from './ChipInput.js';`
8.  got  following syntax error:
![screen shot 2016-09-29 at 7 30 08 pm](https://cloud.githubusercontent.com/assets/998068/18975852/2cfa3a64-867b-11e6-85a3-a464f697c7bd.png)


I'm not sure how to proceed from here.  Thank you in advance!

ps.  Ha, yes, I'm glad I took 'syntacticsugar', but I settled on that after seeing 'argv0' was already taken 😄

**Comment by leMaik** [comment]
@syntacticsugar Oh no! :( Here's how you can test it:

1. Clone this project, let's say into /home/syntacticsugar/material-ui-chip-input
2. In that directory, run `npm i && npm run build`
3. In your project directory, run `npm link /home/syntacticsugar/material-ui-chip-input`
4. ???
5. Profit! :smile: 

But, since my commit from yesterday fixed #14, I'll publish a new release in a moment. So you could just install and test it with the new release.

**Comment by syntacticsugar** [comment]
Hello, I still have the same exact issues, even after 1) going through your steps outlined above, and 2) deleting my node_modules/ and reinstalling your package with `npm i --save material-ui-chip-input`

I'm going to debug this further as there might be something wrong I'm doing on my end regarding making sure I have the latest/fixed version, and I'll get back to you!

**Comment by syntacticsugar** [comment]
__GOT IT TO WORK!__ 🎆 🎆 

After trying different creative approaches and failing (& getting tons of errors in Chrome browser console), I learned that `npm link` creates a symlink that needs to be manually deleted.  I didn't know how, but I ran:

1.  `npm rm --global material-ui-chip-input`
2.  `rm -rf node_modules/material-ui-chip-input`
3.  `npm i --save material-ui-chip-input`
4.   restarted the server
5.   ??? (building suspense...)
5.   prosper!!!  💰 💰 

Hahahahahah!  Duplicate text gone!

Feel free to close this issue :)  Thank you!

ps.  I'm not sure the steps for `npm link` were properly spelled out or executed, in the standalone material-ui-chip-input directory, I had run `git log` and saw I had indeed the latest commit, but I still had duplicate text at the time.  Thank you for publishing to `npm` later though, I think once I properly deleted old references, installed correctly, everything worked.
